### PR TITLE
Filter on alpha3

### LINKED ISSUE #112: Add boundary.country parameter to /search and /reverse endpoints
The geometry of some country polygons makes it difficult to use bbox filtering to exclude neighbouring nations.

It would be nice to have an API param called `alpha3` or similar which excluded results from outside that country.

When used in combination with a coarse geocoder it would be very easy to create a coarse search for a single country using the mapzen service.

### PR DESCRIPTION
**Pull request by orangejulius**
This PR adds an alpha3 parameter to the /search endpoint. It ensures Elasticsearch only returns results matching that alpha3 code via a must filter. A sanitiser checks that the alpha3 parameter is a 3 character string, but doesn't check for validity. Invalid alpha3s (i.e. ones not assigned to any country) will not throw an error, but also won't return any results (obviously). That said I'd like to build an npm module that exports a list of all countries filtered by alpha2, alpha3, etc so that a friendly warning could be returned in the case of an invalid alpha3. But that's for another day.

Fixes #112

### CODE REVIEW

**Comment by riordan** [comment]
Excellent!

**Comment by orangejulius** [comment]
@missinglink, should the analyzer for the admin3 field be changed to keyword? anything other than an exact match is not interesting

File: sanitiser/_alpha3.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+// validate inputs, convert types and apply defaults
+function sanitize( req ){
```
Comment (dianashk) [inline sanitiser/_alpha3.js]: this needs to now match the new convention. please merge master into the branch. sanitize functions now take `(raw, clean)`

File: sanitiser/_alpha3.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  var params = req.query;
+  req.clean = req.clean || {};
+
+  if (params === undefined || params.alpha3 === undefined) {
```
Comment (dianashk) [inline sanitiser/_alpha3.js]: shouldn't the param name be `boundary.country`?

Comment (orangejulius) [inline sanitiser/_alpha3.js]: Really old code in this PR, sorry

File: sanitiser/_alpha3.js (hunk 1 above)
Comment (orangejulius) [inline sanitiser/_alpha3.js]: Really old code in this PR, sorry

**Comment by orangejulius** [comment]
Really old code in this PR

**Pull request closed without merging**